proto/execution_stats.proto

```diff
@@ -127,6 +126,13 @@ message Execution {
   // retried internally, this Execution will reflect the latest execution
   // attempt.
   string execution_id = 14;
+
+  // Information about Bazel build target which created this Action's Execution.
+  // Note that these fields could be empty if the client does not send a
+  // RequestMetadata with these fields populated.
+  string target_id = 15;
```

Comment:
I generally like the idea of this, but I think the reason we've avoided doing this in the past is that the executions table is already huge (both in number of rows and overall size), and this would add 3 large variable length strings which would further exacerbate the problem.

This feels like something better suited either for Clickhouse or if we ever end up storing the executions for an invocation in a GCS blob like we've talked about.

Reply:
so i agree that this is a lot of data to jam into mysql when we would probably never do a lookup based on it, but just to be clear, we basically never do lookups unless we're keying off of execution id or invocation id, right?  is there some case i'm forgetting about where having very-wide rows in the executions table actually slows things down?  does it matter for insert speed?

Reply:
Was mostly just thinking for size, and us moving away from using mysql for execution metadata. Our MySQL DB is over 5TB, and I think over half of that is the executions table (though I may be wrong).

Don't get me wrong, I would LOVE to have this data more easily available. If @tylerwilliams is cool with it, I am.

Reply:
wrt to the MySQL execution tables sizes: we probably could look into the execution janitor a bit more and see if it's keeping up with the current rate of executions.

Reply:
I think even with our janitor approach, there is fragmentation happening and we're not recouping space on disk.

I'd prefer not writing more data to the mysql executions table, but I think putting it in clickhouse would be fine?

Reply:
I pushed a change that include the following:

a. Write the metadata to Redis instead of MySQL
b. Flush the data from Redis to ClickHouse when the execution completes
c. When GetExecution is called, fetch the metadata from Redis for incompleted actions and Clickhouse for completed actions